### PR TITLE
ac-55 hook up authentication to a new database schema

### PR DESCRIPTION
**Pull request by ExtraGravity**
Remove old user model.
Authentication is very similar, it just gets the user information
from the new database now. Function were written in the database layer
to get this information.

### CODE REVIEW

File: src/database/Person/index.ts [hunk 1]
```diff
@@ -1,37 +1,100 @@
...
+    .join('people', 'people.personcategoryid', 'personcategories.personcategoryid')
+    .where({ personuuid });
+  },
+};
+
+/** Build a UserType object from a string */
+function buildUserType(type: string): UserType {
+  switch (type) {
+    case UserType.ADMIN:
+      return UserType.ADMIN;
```
Comment (beaesguerra) [inline src/database/Person/index.ts]: can you put brackets around each case?

File: src/database/Person/index.ts [hunk 2]
```diff
@@ -1,37 +1,100 @@
...
+    case UserType.MONITOR:
+      return UserType.MONITOR;
+    case UserType.TRADER:
+      return UserType.TRADER;
```
Comment (beaesguerra) [inline src/database/Person/index.ts]: you can use a default case to log the error

File: src/database/Person/index.ts [hunk 3]
```diff
@@ -1,37 +1,100 @@
...
+  }
+  logger.error('Unrecognized user type', type);
+}
+
+/** Get attributes for a user necessary for a login from a username */
+export async function findUser(username: string): Promise<User> {
+  logger.info('Person/index.ts findUser');
```
Comment (beaesguerra) [inline src/database/Person/index.ts]: this can be removed?

File: src/database/Person/index.ts [hunk 4]
```diff
@@ -1,37 +1,100 @@
...
+  logger.info('uuid, ', uuid);
+
+  const hashResults = await execute<AttrValue[]>(builders.getAttr(uuid, 'hash'));
+  if (hashResults.length === 0) {
+    throw new Error(username + ' has no password set!');
+  }
+  logger.info('hash, ', hashResults[0].attrvalue);
+
+  const userTypeResults = await execute<PersonCategoryName[]>(builders.getPersonCategory(uuid));
+  if (userTypeResults.length === 0) {
```
Comment (beaesguerra) [inline src/database/Person/index.ts]: is it possible for a user to not be in a category? no right?

File: src/database/Person/index.ts [hunk 5]
```diff
@@ -1,37 +1,100 @@
...
+/** Get attributes for a user necessary for a login from a username */
+export async function findUser(username: string): Promise<User> {
+  logger.info('Person/index.ts findUser');
+
+  const uuidResults = await execute<PersonUuid[]>(builders.findUserUuid(username));
+  if (uuidResults.length === 0) {
+    throw new Error('Username ' + username + ' not found');
+  }
+  const uuid = uuidResults[0].personuuid;
+  logger.info('uuid, ', uuid);
```
Comment (beaesguerra) [inline src/database/Person/index.ts]: do we need this? if we want to trackk users that are logging in, we can make this message more meaningful

File: src/database/Person/index.ts [hunk 6]
```diff
@@ -1,37 +1,100 @@
...
+function buildUserType(type: string): UserType {
+  switch (type) {
+    case UserType.ADMIN:
+      return UserType.ADMIN;
+    case UserType.MONITOR:
+      return UserType.MONITOR;
+    case UserType.TRADER:
+      return UserType.TRADER;
+  }
+  logger.error('Unrecognized user type', type);
```
Comment (beaesguerra) [inline src/database/Person/index.ts]: do we need to throw error?

File: src/database/Person/index.ts [hunk 7]
```diff
@@ -1,37 +1,100 @@
...
+  const uuidResults = await execute<PersonUuid[]>(builders.findUserUuid(username));
+  if (uuidResults.length === 0) {
+    throw new Error('Username ' + username + ' not found');
+  }
+  const uuid = uuidResults[0].personuuid;
+  logger.info('uuid, ', uuid);
+
+  const hashResults = await execute<AttrValue[]>(builders.getAttr(uuid, 'hash'));
+  if (hashResults.length === 0) {
+    throw new Error(username + ' has no password set!');
```
Comment (beaesguerra) [inline src/database/Person/index.ts]: log this error? this seems like a pretty bad scenario we would want to know about

**Review by beaesguerra — COMMENTED** [review]
questions

File: src/database/Person/index.ts (hunk 4 above)
Comment (ExtraGravity) [reply src/database/Person/index.ts]: not possible, I think it's enforced by the db.

File: src/database/Person/index.ts (hunk 7 above)
Comment (ExtraGravity) [reply src/database/Person/index.ts]: no I think if it an error gets thrown, it either gets properly dealt with higher up, or it will logged as a crash.

**Review by beaesguerra — APPROVED** [review]

**Pull request merged**